system/lib/wasmfs/wasmfs.cpp

```diff
@@ -139,51 +134,50 @@ __wasi_fd_t __syscall_open(long pathname, long flags, long mode) {
     canWrite = true;
   }
 
-  std::string interim = "";
   std::vector<std::string> pathParts;
 
-  auto newPathName = (const char*)pathname;
-  // TODO: Support relative paths
-  for (int i = 0; newPathName[i] != '\0'; i++) {
-    if (newPathName[i] != '/') {
-      interim += newPathName[i];
-    } else {
-      // Avoid adding empty string
-      if (i > 0) {
-        pathParts.push_back(interim);
-      }
-      interim = "";
-    }
+  auto newPathName = (char*)pathname;
+
+  // TODO: Support relative paths. i.e. specify cwd if path is relative.
+  // TODO: Other path parsing edge cases.
+  char* current;
+
+  current = strtok(newPathName, "/\n");
```

Comment:
Best not to use `strtok` here because it will modify its input.

Reply:
Added a strcopy